pagefaultgames/pokerogue
# [Refactor][Move][Ability] Fixing up MoveTypeChangeAttr implementation

## What are the changes the user will see?
The user will be able to paralyze ground-type pokemon and unable to paralyze ghost-types if the Pokemon involved has the ability Normalize. If the user with a non-Normalize Move Type Change ability uses a normal type multi-hit move against a ghost user, it will be able to hit ghost types multiple times.

## Why am I making these changes?
Just a small improvement to code logic. Fixing up issue #3570 

## What are the changes from a developer perspective?
Abilities that change the types of moves are applied when executing status moves.
Replaced StatusMoveTypeImmunity attribute with RespectTypeImmunity attribute. This checks if an opposing Pokemon has a type that is immune to the current type of the move. 

### Screenshots/Videos
https://github.com/user-attachments/assets/f38b6a90-36fc-43cb-ad96-3cc8d8afda14

https://github.com/user-attachments/assets/4553e8fe-4408-49bd-b7d2-e71cb893a627

https://github.com/user-attachments/assets/190b73aa-ab79-486e-9769-8ee74d436b38

### Screenshots/Videos of Aerilate + Population Bomb

https://github.com/user-attachments/assets/8dc5a389-6811-4646-be01-a42a4dfc00f0

## How to test the changes?
Use Thunder Wave with a Pokemon with the ability Normalize --> Ground Types no longer immune, Ghost Types immune
Use Thunder Wave with a Pokemon without the above ability --> Ground Types immune, Ghost Types not immune

## Checklist
